tests/integration/metrics_tests/test_metrics.py

```diff
@@ -38,11 +38,10 @@ async def get_address(ops_test: OpsTest, app_name=DATABASE_APP_NAME, unit_num=0)
 
 async def verify_endpoints(ops_test: OpsTest, app_name=DATABASE_APP_NAME):
     """Verifies mongodb endpoint is functional on a given unit."""
-    http = urllib3.PoolManager()
-
     for unit_id in range(len(ops_test.model.applications[app_name].units)):
-        app_address = await get_address(ops_test=ops_test, app_name=app_name, unit_num=unit_id)
-        mongo_resp = http.request("GET", f"http://{app_address}:{MONGODB_EXPORTER_PORT}/metrics")
+        http = urllib3.PoolManager()
+        unit_address = await get_address(ops_test=ops_test, app_name=app_name, unit_num=unit_id)
+        mongo_resp = http.request("GET", f"http://{unit_address}:{MONGODB_EXPORTER_PORT}/metrics")
```

Comment:
And we should probably add the `assert` in the loop
```suggestion
            unit_address = await get_address(ops_test=ops_test, app_name=app_name, unit_num=unit_id)
            mongo_resp = http.request("GET", f"http://{unit_address}:{MONGODB_EXPORTER_PORT}/metrics")
            assert mongo_resp.status_code == 200
```